src/openqaoa-core/openqaoa/problems/kcolor.py

```diff
@@ -0,0 +1,194 @@
+        mdl.minimize(cost)
+        return mdl
+
+    def classical_solution(self, string: bool = False):
+        """
+        Return the classical solution of the k-color problem
+
+        Parameters
+        ----------
+        string : bool, optional
+            If the solution is returned as a string. The default is False.
+
+        Returns
+        -------
+        solution : Union[str, dict]
+            The classical solution of the specific problem as a string or a dict.
+        """
+        mdl = self.docplex_model
+        solution = mdl.solve()
+
+        if solution is None:
+            raise ValueError("No valid coloring found.")
+
+        if string:
+            coloring = ""
+        else:
+            coloring = {}
+        for var in mdl.iter_binary_vars():
+            if string:
+                coloring += str(int(round(solution.get_value(var), 1)))
+            else:
+                coloring[var.name] = int(round(solution.get_value(var), 1))
+        return coloring
+
+    @property
+    def qubo(self):
+        """
+        Returns the QUBO encoding of this problem.
+
+        Returns
+        -------
+            The QUBO encoding of this problem.
+        """
+        cplex_model = self.docplex_model
+        qubo_docplex = FromDocplex2IsingModel(cplex_model).ising_model
+
+        return QUBO(
+            self.G.number_of_nodes(),
+            qubo_docplex.terms + [[]],
+            qubo_docplex.weights + [qubo_docplex.constant],
+            self.problem_instance,
+        )
+
+    def plot_solution(self, solution: Union[dict, str], ax=None, colors=None):
+        """
+        A visualization method for the k-color solution.
+
+        Parameters
+        ----------
+        solution : Union[dict, str]
+            The solution of the specific k-color problem as a string or dictionary.
+        ax : matplotlib axes, optional
+            The default is None.
+
+        Returns
+        -------
+        fig : matplotlib.pyplot.Figure()
+            The graph visualization of the solution.
+        """
+        if isinstance(solution, str):
+            sol = {}
+            for vertex, color in solution:
+                sol[vertex] = color
+            solution = sol
+
+        if ax is None:
+            fig, ax = plt.subplots()
+        else:
+            fig = None
+
+        colors = colormaps["jet"] if colors is None else colors
+        print([(col+1)/(self.k+1) for vertex in self.G for col in range(self.k) if solution[f"x_{vertex}_{col}"] == 1])
+        pos = nx.circular_layout(self.G)
+        nx.draw(
+            self.G,
+            pos=pos,
+            node_color=[colors((col+1)/(self.k+1)) for vertex in self.G for col in range(self.k) if solution[f"x_{vertex}_{col}"] == 1],
+            edge_color='black',
+            with_labels=True,
+            ax=ax,
+            alpha=0.85,
+            **{"edgecolors":"black"}
+        )
+        return fig
```

Comment:
When I do:
```
prob = KColor.random_instance()
prob.plot_solution(prob.classical_solution())
```
OR
```
prob = KColor.random_instance()
class_sol  = prob.classical_solution()
prob.plot_solution(class_sol)
```
I get 2 graph outputs instead of 1 in a jupyter notebook.